agent/config/config_test.go

```diff
@@ -33,6 +33,20 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestDefaultValueOfDisconnectModeEnabled(t *testing.T) {
+
+	assert.False(t, GetDisconnectModeEnabled(), "Wrong default value for disconnectModeEnabled")
+}
+
+func TestGetAndSetDisconnectModeEnabled(t *testing.T) {
+
+	SetDisconnectModeEnabled(true)
+	assert.True(t, GetDisconnectModeEnabled(), "GetDisconnectModeEnabled not working as expected")
```

Comment:
nit - Maybe we can add expected value in the error message (here and below as well)?

Reply:
yeah you're right! will add that